### PR TITLE
Add IdBasedSearchFetcher to jstor

### PR DESCRIPTION
**Pull request by joethei**
As discussed in https://github.com/JabRef/jabref/issues/6627#issuecomment-733514445

- [ ] Change in CHANGELOG.md described (if applicable)
- [X] Tests created for changes (if applicable)
- [X] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [X] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/logic/importer/fetcher/JstorFetcher.java [hunk 1]
```diff
@@ -82,6 +84,46 @@ public URL getURLForQuery(ComplexSearchQuery query) throws URISyntaxException, M
...
+                return new URL(start + identifier);
+            }
+            // else use default doi start.
+            return new URL(start + "10.2307/" + identifier);
+        } catch (IOException e) {
+            throw new FetcherException("could not construct url for jstor", e);
+        }
+    }
+
+    // overwriting default behaviour as that seems to always results in a 403
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/logic/importer/fetcher/JstorFetcher.java:108]: What did you had to change relative to the standard implementation?

Comment (joethei) [reply src/main/java/org/jabref/logic/importer/fetcher/JstorFetcher.java:108]: getParser was already written to only work for search so I simply overwrote this method to use a bibtex parser.
I did have another look at the code, removed this part and added the required logic to getParser.

**Review by tobiasdiez — APPROVED** [review]
Thanks for your contribution! Looks good to me

File: src/main/java/org/jabref/logic/importer/fetcher/JstorFetcher.java [hunk 2]
```diff
@@ -133,4 +169,9 @@ public String getName() {
     public TrustLevel getTrustLevel() {
         return TrustLevel.META_SEARCH;
     }
+
+    @Override
+    public void doPostCleanup(BibEntry entry) {
+        // do nothing
+    }
```
Comment (DominikVoigt) [inline src/main/java/org/jabref/logic/importer/fetcher/JstorFetcher.java:176]: Why did you overwrite this?

**Review by DominikVoigt — CHANGES_REQUESTED** [review]
Hey,
Thanks for your contribution!
I only have one question otherwise LGTM!

File: src/main/java/org/jabref/logic/importer/fetcher/JstorFetcher.java (hunk 2 above)
Comment (joethei) [reply src/main/java/org/jabref/logic/importer/fetcher/JstorFetcher.java:176]: as both SearchBased and IdBased Fetchers have a doPostCleanup default impl, the compiler is not sure which one to call

**Review by DominikVoigt — APPROVED** [review]
Thanks for the clarification! :)

**Pull request merged**